### PR TITLE
[SYCL] Deprecate some SYCL 1.2.1 APIs in SYCL2020 mode

### PR DESCRIPTION
**Pull request by alexbatashev**
Also cleanup use of deprecated APIs in runtime source code.

### CODE REVIEW

File: sycl/include/CL/sycl/aliases.hpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ using half = cl::sycl::detail::half_impl::half;
 
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
-using byte = std::uint8_t;
+using byte __SYCL2020_DEPRECATED("use std::byte instead") = std::uint8_t;
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/aliases.hpp:74]: "std::byte" is C++17: are we requiring it for SYCL now?

File: sycl/include/CL/sycl/device.hpp [hunk 2]
```diff
@@ -161,17 +161,18 @@ class __SYCL_EXPORT device {
   ///
   /// \param extension_name is a name of queried extension.
   /// \return true if SYCL device supports the extension.
-  bool has_extension(const string_class &extension_name) const;
+  __SYCL2020_DEPRECATED("use device aspects APIs instead")
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/device.hpp:164]: better to give the right name in the message, is it "has_aspect"?

File: sycl/include/CL/sycl/handler.hpp [hunk 3]
```diff
@@ -1294,6 +1294,7 @@ class __SYCL_EXPORT handler {
   /// \param KernelFunc is a SYCL kernel function.
   template <typename KernelName = detail::auto_name, typename KernelType,
             int Dims>
+  __SYCL2020_DEPRECATED("offsets are deprecated in SYCL2020")
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/handler.hpp:1297]: can we say what to use instead?

File: sycl/include/CL/sycl/INTEL/online_compiler.hpp [hunk 4]
```diff
@@ -61,7 +61,7 @@ class device_arch {
 class online_compile_error : public sycl::exception {
 public:
   online_compile_error() = default;
-  online_compile_error(const string_class &Msg) : sycl::exception(Msg) {}
+  online_compile_error(const std::string &Msg) : sycl::exception(Msg) {}
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/INTEL/online_compiler.hpp:64]: is this a NOP change? does it not affect ABI?

**Comment by smaslov-intel** [comment]
Can you break this into multiple change-sets? At least separate renaming of "*_class" from deprecating APIs

File: sycl/include/CL/sycl/aliases.hpp (hunk 1 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/aliases.hpp:74]: For SYCL2020 we do. The warning won't be shown for older editions.

File: sycl/include/CL/sycl/device.hpp (hunk 2 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/device.hpp:164]: It is just `has`.

File: sycl/include/CL/sycl/handler.hpp (hunk 3 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/handler.hpp:1297]: Nothing. There's no replacement in SYCL2020.

File: sycl/include/CL/sycl/INTEL/online_compiler.hpp (hunk 4 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/INTEL/online_compiler.hpp:64]: Aliases do no participate in name mangling, so this change won't affect ABI

**Review by smaslov-intel — APPROVED** [review]

**Comment by bader** [comment]
@alexbatashev, please, address pre-commit failures.

**Comment by dm-vodopyanov** [comment]
@bader, Jenkins/Precommit is passing now, also fixed merge conflicts.

**Pull request merged**